includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -865,7 +866,13 @@ public function query( $query_vars ) {
 			$query = new WP_Query( $args );
 		}
 
-		$orders = ( isset( $query_vars['return'] ) && 'ids' === $query_vars['return'] ) ? $query->posts : array_filter( array_map( 'wc_get_order', $query->posts ) );
+		if ( isset( $query_vars['return'] ) && 'ids' === $query_vars['return'] ) {
+			$orders = $query->posts;
+		} else {
+			update_post_caches( $query->posts ); // We already fetching posts, might as well hydrate some caches.
```

Comment:
Do you think we might want to set the `metadata` parameter to `false` here since we're not using WordPress' meta functions?